brage-import/src/test/java/no/sikt/nva/brage/migration/merger/PublicationInstanceMergerTest.java

```diff
@@ -0,0 +1,67 @@
+package no.sikt.nva.brage.migration.merger;
+
+import static no.unit.nva.hamcrest.DoesNotHaveEmptyValues.doesNotHaveEmptyValues;
+import static no.unit.nva.model.testing.PublicationGenerator.randomPublication;
+import static org.hamcrest.MatcherAssert.assertThat;
+import java.util.Set;
+import java.util.stream.Stream;
+import no.sikt.nva.brage.migration.model.PublicationRepresentation;
+import no.sikt.nva.brage.migration.record.Record;
+import no.unit.nva.model.Publication;
+import no.unit.nva.model.exceptions.InvalidIsbnException;
+import no.unit.nva.model.exceptions.InvalidIssnException;
+import no.unit.nva.model.exceptions.InvalidUnconfirmedSeriesException;
+import no.unit.nva.model.instancetypes.degree.DegreeBachelor;
+import no.unit.nva.model.instancetypes.degree.DegreeBase;
+import no.unit.nva.model.instancetypes.degree.DegreeLicentiate;
+import no.unit.nva.model.instancetypes.degree.DegreeMaster;
+import no.unit.nva.model.instancetypes.degree.DegreePhd;
+import no.unit.nva.model.instancetypes.degree.OtherStudentWork;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class PublicationInstanceMergerTest {
+
+    public static Stream<Arguments> emptyDegreeSupplier() {
+        return Stream.of(Arguments.of(new DegreePhd(null, null, Set.of())),
+                         Arguments.of(new DegreeBachelor(null, null)),
+                         Arguments.of(new DegreeMaster(null, null)),
+                         Arguments.of(new DegreeLicentiate(null, null)),
+                         Arguments.of(new OtherStudentWork(null, null)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("emptyDegreeSupplier")
+    void shouldUseExistingPublicationInstanceWhenNewPublicationInstanceIsEmpty(DegreeBase degreeBase)
```

Comment:
Would prefer it if you did this from a TDD standpoint, we have som many rules to when two publications are not considered matching, so make this test execute from the handler and not from this class.

Reply:
Legger til en test på handler nivå:

```
    @ParameterizedTest
    @MethodSource("emptyDegreeSupplier")
    void shouldUpdateExistingPublicationByFillingUpInstanceTypeEmptyValues(PublicationInstance<?> publicationInstance, Type type) throws IOException {
        var cristinIdentifier = "1234";
        var generator = generateBrageRecordAndPersistDuplicateByCristinIdentifier(publicationInstance, type, cristinIdentifier);
        var s3Event = createNewBrageRecordEvent(generator.getBrageRecord());
        var updatedPublicationInstance = handler.handleRequest(s3Event, CONTEXT)
                                             .publication().getEntityDescription().getReference().getPublicationInstance();

        assertThat(updatedPublicationInstance,
                   doesNotHaveEmptyValuesIgnoringFields(Set.of(".pages.introduction", ".pages.illustrated")));
    }
```